File: frontend/src/types/people.ts
export type SwipeDirection = 'ALIGN' | 'OPPOSE';

export interface DeckItem {
  id: string;
  platform: string;
  handle: string;
  display_name: string | null;
  bio: string | null;
  score: number;
  owner_user_id: string | null;
  status: string;
  week_delta?: number | null;
}

export interface DeckResponse {
  items: DeckItem[];
  next_cursor: number;
  has_more: boolean;
}

export interface SwipeResult {
  account_id: string;
  direction: SwipeDirection;
  delta: number;
  new_account_score: number;
}

File: backend/app/schemas/people.py
from __future__ import annotations

from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.models.enums import AccountStatus, SwipeDirection


class TrendingAccountOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    platform: str
    handle: str
    display_name: str | None
    score: float
    owner_user_id: UUID | None


class DeckAccountOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    platform: str
    handle: str
    display_name: str | None
    bio: str | None
    score: float
    owner_user_id: UUID | None
    status: AccountStatus
    week_delta: float | None = None


class SwipeRequest(BaseModel):
    direction: SwipeDirection


class SwipeResponse(BaseModel):
    account_id: UUID
    direction: SwipeDirection
    delta: float
    new_account_score: float


class TrendingResponse(BaseModel):
    items: list[TrendingAccountOut]


class DeckResponse(BaseModel):
    items: list[DeckAccountOut]
    next_cursor: int
    has_more: bool
